x-pack/plugins/reporting/server/types.ts

```diff
@@ -117,3 +118,10 @@ export interface ReportingRequestHandlerContext {
  * @internal
  */
 export type ReportingPluginRouter = IRouter<ReportingRequestHandlerContext>;
+
+type Url = string;
+type UrlLocatorTuple = [url: Url, locatorParams: LocatorParams];
+/**
+ * @internal
+ */
+export type UrlOrUrlLocatorTuple = Url | UrlLocatorTuple;
```

Comment:
I think we want to add this to `x-pack/plugins/reporting/common/types.ts` as there seems to be nothing server-specific to this.